Avoid preserving permissions in `file(COPY ...)`

In the following code from `source/utils/adios_iotest/CMakeLists.txt`:
```
file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/iotest-config
  DESTINATION ${PROJECT_BINARY_DIR}
)
```
This file(COPY ...) command attempt to preserve permissions on the copied files, which implicitly requires `chmod` access. This causes build failures when compiling ADIOS2 on a shared server where chmod is not permitted on certain filesystems. The error message is like:
```
CMake Error at source/utils/adios_iotest/CMakeLists.txt:10 (file):
  file COPY cannot set permissions on
  "/users/myusername/_work/workspace//build-ADIOS2-cuda/iotest-config":
  Operation not permitted.
```
Is preserving permissions necessary for this copy? Would it be possible to add the `NO_SOURCE_PERMISSIONS` option to avoid this requirement?

--- a/source/utils/adios_iotest/CMakeLists.txt
+++ b/source/utils/adios_iotest/CMakeLists.txt
@@ -7,10 +7,6 @@
 # accompanying file Copyright.txt for details.
 #------------------------------------------------------------------------------#
 
-file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/iotest-config
-  DESTINATION ${PROJECT_BINARY_DIR}
-)
-
 add_executable(adios_iotest settings.cpp decomp.cpp processConfig.cpp ioGroup.cpp stream.cpp adiosStream.cpp adios_iotest.cpp)
 target_link_libraries(adios_iotest adios2::cxx_mpi MPI::MPI_CXX adios2_core_mpi)
 if(WIN32)

